lib/tests/formulae.rb

```diff
@@ -225,19 +225,66 @@ def livecheck(formula)
         return if livecheck_step.failed?
         return unless livecheck_step.output?
 
-        livecheck_info = JSON.parse(livecheck_step.output)
-        return unless livecheck_info.first["version"]["newer_than_upstream"]
+        livecheck_info = JSON.parse(livecheck_step.output)&.first
+        if livecheck_info.blank?
+          # This warning shouldn't appear under normal circumstances. If we
+          # encounter this, something unexpectedly changed/broke in livecheck.
+          no_info_msg = "The data from livecheck is unexpectedly empty"
+          if ENV["GITHUB_ACTIONS"].present?
+            puts GitHub::Actions::Annotation.new(
+              :warning,
+              no_info_msg,
+              title: "#{formula}: livecheck data is empty",
+              file:  formula.path.to_s.delete_prefix("#{repository}/"),
+            )
+          else
+            opoo no_info_msg
+          end
+        end
+
+        if livecheck_info["status"] == "error"
+          error_msg = if livecheck_info["messages"].present? && livecheck_info["messages"].length.positive?
+            livecheck_info["messages"].join("\n")
+          else
+            # An error message should always be provided alongside an "error"
+            # status but this is a failsafe
+            "Error encountered (no message provided)"
+          end
+
+          if ENV["GITHUB_ACTIONS"].present?
+            puts GitHub::Actions::Annotation.new(
+              :warning,
+              error_msg,
+              title: "#{formula}: livecheck error",
+              file:  formula.path.to_s.delete_prefix("#{repository}/"),
+            )
+          else
+            opoo error_msg
+          end
+        end
+
+        return if livecheck_info["version"].blank?
```

Comment:
Is this needed? Sorry, I'm just being very cautious about silently ignoring things that should be flagged as an issue.

If this should never happen: allowing it to naturally error if it does is fine.

Reply:
Something like this is still needed with the current setup, as there are a variety of non-error statuses related to skip conditions (e.g., "skipped", "deprecated", "disabled", "versioned", "discontinued", "latest", "unversioned"). That is to say, either `status` or `version` will be present but `status` will not only be `"error"`.

If we want to avoid `return if livecheck_info["version"].blank?`, we would need to replace it with something like `return if livecheck_info["status"].present?`. In this scenario, if `livecheck_info["version"]` is ever blank it would be a genuine bug and would error like we're currently seeing. Does this work for you?

Reply:
Ok that makes sense. Yes I think `return if livecheck_info["status"].present?` is semantically clearer. I didn't understand the logic before but I do now.